app-experimental.tex

```diff
@@ -2395,3 +2395,219 @@ \section{Distributed Capabilities For Peripherals And Accelerators} % <<<
+A SCID may consist of the corresponding ACID added to other bits in order to create a virtual identifier or can be fully independent of the ACID is mapped to.
+
+Compartment IDs come with two new instructions.
+One instruction for reading the CID into a general purpose register and one instruction for setting the CID of a capability.
+Like capabilities themselves, CIDs are not considered secret.
+Thereforem, the CID reading instruction is not privileged, but the writing instruction is restricted (see below).
+We suggest encodings for the instructions in order to demonstrate the adaptability of CID to the currently existing CHERI-RISC-V ISA.
+
+\texttt{CSetCID cd, rs0}:\\
+set the CID in cd to the value in rs0.
+This instruction needs the PERMIT\_SET\_CID bit set, otherwise it throws an exception.
+This instruction uses the ACID\_LENGTH lower bits of rs0 and ignores the upper bits.
+
+Possible encoding (31:0): 0x7f, 0x19, rs0, 0x0, cd, 0x5b (assign a random free funct5)
+
+\texttt{CGetCID rd, cs1}:\\
+extract the CID of cs1 and store it in rd
+
+Preliminary encoding (31:0): 0x7f, 0x1, cs1, 0x0, rd, 0x5b (this is the same encoding as CGetType, which will be obsolete with this proposal)
+
+The PERMIT\_SET\_CID bit is a hardware permission bit that, if set in a PCC, allows manipulating the ACID.
+Like all hardware permission bits in CHERI, PERMIT\_SET\_CID is constrained by monotonicity.
+A code capability with this bit set should only be available to supervisor code.
+
+
+\subsection{Sealing}
+
+The current CID (the CID of the current compartment) is determined by the CID in the PCC, which is also a capability.
+If the PCC changes and the new PCC has a different CID, this constitutes a compartment change (see Section~\ref{subsec:comp_change}).
+
+With an ACID being in all capabilities, we can establish a concept referred to as CID Sealing.
+All capabilities are implicitly sealed by their CID.
+This means that they can only be unsealed if their CID matches the \texttt{PCC.CID}.
+An implicitly sealed capability can be inspected (its fields can be read), but it cannot be manipulated nor can it be used to reference memory.
+
+CID sealing allows a register file to hold capabilities from different compartments without allowing capability leaks.
+The currently executing compartment can only make use of its own capabilities.
+
+In this proposal, CID sealing completely replaces the sealing mechanism currently present in CHERI.
+A CID sealed capability can only be manipulated by the compartment it is owned by.
+Therefore, it can be securely handed out to other compartments, e.g., as a code pointer back.
+
+However, CID sealing can also co-exist with current sealing mechanisms in place, e.g., sentries.
+In comparison to conventional CHERI-RISC-V sealing, CID sealing cannot produce code pointers that are immutable within a compartment, but only across compartments.
```

Comment:
Discussions about orthogonal sentry bit...

Reply:
Yes, that is good. I added a paragraph in the storage sub section and referred to it.